cargo-near/src/main.rs

```diff
@@ -1,27 +1,106 @@
+pub mod common;
+pub mod types;
+pub mod util;
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(context = near_cli_rs::GlobalContext)]
+struct Cmd {
+    #[interactive_clap(subcommand)]
+    opts: Opts,
+}
+
+#[derive(Debug, EnumDiscriminants, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(context = near_cli_rs::GlobalContext)]
+#[strum_discriminants(derive(EnumMessage, EnumIter))]
+#[interactive_clap(disable_back)]
+/// Near
+pub enum Opts {
+    #[strum_discriminants(strum(message = "near"))]
+    /// Which cargo extension do you want to use?
+    Near(NearArgs),
+}
+
+#[derive(Debug, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(context = near_cli_rs::GlobalContext)]
+pub struct NearArgs {
+    #[interactive_clap(subcommand)]
+    pub cmd: NearCommand,
+}
+
+#[derive(Debug, EnumDiscriminants, Clone, interactive_clap::InteractiveClap)]
+#[interactive_clap(context = near_cli_rs::GlobalContext)]
+#[strum_discriminants(derive(EnumMessage, EnumIter))]
+#[interactive_clap(disable_back)]
+/// What are you up to? (select one of the options with the up-down arrows on your keyboard and press Enter)
+pub enum NearCommand {
+    #[strum_discriminants(strum(
+        message = "build   -   Build a NEAR contract and optionally embed ABI"
+    ))]
+    /// Build a NEAR contract and optionally embed ABI
+    Build(self::build_command::BuildCommand),
+    #[strum_discriminants(strum(message = "abi     -   Generates ABI for the contract"))]
+    /// Generates ABI for the contract
+    Abi(self::abi_command::AbiCommand),
+}
+
+fn main() -> CliResult {
     env_logger::init();
 
     match env::var("NO_COLOR") {
         Ok(v) if v != "0" => colored::control::set_override(false),
         _ => colored::control::set_override(atty::is(atty::Stream::Stderr)),
     }
 
-    let Opts::Near(args) = Opts::parse();
-    match cargo_near::exec(args.cmd) {
-        Ok(()) => {}
-        Err(err) => {
-            let err = format!("{:?}", err);
-            let mut err_lines = err.lines();
-            eprintln!(" {} {}", "✗".bright_red().bold(), err_lines.next().unwrap());
-            for line in err_lines {
-                eprintln!("   {}", line);
+    let config = near_cli_rs::common::get_config_toml()?;
+
+    color_eyre::install()?;
+
+    let cli = match Cmd::try_parse() {
+        Ok(cli) => cli,
+        Err(error) => error.exit(),
+    };
+
+    let global_context = near_cli_rs::GlobalContext {
+        config,
+        offline: false,
+    };
+
+    loop {
+        match <Cmd as interactive_clap::FromCli>::from_cli(
+            Some(cli.clone()),
+            global_context.clone(),
+        ) {
+            interactive_clap::ResultFromCli::Ok(cli_cmd)
+            | interactive_clap::ResultFromCli::Cancel(Some(cli_cmd)) => {
+                eprintln!(
+                    "Your console command:\n{} {}",
+                    std::env::args().next().as_deref().unwrap_or("./cargo-near"),
```

Comment:
We need to make sure that if users call `cargo near build` we don't suggest them to use `cargo-near near build`. I feel it is currently the case, correct me if I am wrong

Reply:
Done